test/strscan/test_stringscanner.rb

```diff
@@ -206,6 +206,19 @@ def test_pos_unicode
     assert_equal 11, s.charpos
   end
 
+  def test_charpos_avoid_segfault
```

Comment:
```suggestion
  def test_charpos_not_use_byteslice
```

Reply:
Thanks for the suggestion! Applied in https://github.com/ruby/strscan/pull/20/commits/ab32d1767c4a1d5c1ccbc05ff8e77161d74be757